utils/appliance.py

```diff
@@ -444,8 +444,12 @@ def rest_api(self):
     @lazycache
     def address(self):
         # If address wasn't set in __init__, use the hostname from base_url
-        parsed_url = urlparse(self.url or store.base_url)
-        return parsed_url.netloc
+        if getattr(self, "_url", None) is not None:
+            parsed_url = urlparse(self.url)
+            return parsed_url.netloc
+        else:
+            parsed_url = urlparse(store.base_url)
+            return parsed_url.netloc
```

Comment:
```python
if getattr(self, "_url", None) is None:
  parsed_url = urlparse(store.base_url)
else:
  parsed_url = urlparse(self.url)
return parsed_url
```

That said, it seems like the problem here is `self.url` not always doing the right thing. Seems like that should be the place to look into store.base_url and set the address properly.